lib/httplib/httplib.go

```diff
@@ -69,6 +70,19 @@ func MakeStdHandler(fn StdHandlerFunc) http.HandlerFunc {
 	}
 }
 
+// WithCSRFProtection ensures that request to unauthenticated API is checked against CSRF attacks
+func WithCSRFProtection(fn HandlerFunc) httprouter.Handle {
+	hanlderFn := MakeHandler(fn)
+	return func(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
+		err := csrf.VerifyHTTPHeader(r)
+		if err != nil {
+			trace.WriteError(w, trace.AccessDenied("failed to validate CSRF token", err))
```

Comment:
no need to explain why access is denied, keep it obscure to make it harder for the attacker to guess why the request has failed.